support/cas-server-support-pm-webflow/src/main/java/org/apereo/cas/pm/web/flow/actions/VerifySecurityQuestionsAction.java

```diff
@@ -46,9 +46,9 @@ protected Event doExecute(final RequestContext requestContext) throws Exception
         
         final Map<String, String> questions = passwordManagementService.getSecurityQuestions(username);
         final AtomicInteger i = new AtomicInteger(0);
-        final long c = questions.values().stream().filter(v -> {
+        final long c = questions.entrySet().stream().filter(e -> {
             final String answer = request.getParameter("q" + i.getAndIncrement());
-            return answer.equals(v);
+            return passwordManagementService.checkSecurityQuestionAnswer(username, e.getKey(), e.getValue(), answer);
```

Comment:
Rename this to `isValidAnswer()` please, or more verbosely, `isValidSecurityQuestionAnswer()`.